openshift_cli_installer/libs/clusters/ipi_cluster.py

```diff
@@ -111,6 +116,12 @@ def _create_install_config_file(self):
         with open(os.path.join(self.cluster_info["cluster-dir"], "install-config.yaml"), "w") as fd:
             fd.write(yaml.dump(cluster_install_config))
 
+    def _set_docker_config_file(self):
+        self.ipi_docker_config_dir = tempfile.mkdtemp()
+        os.environ["DOCKER_CONFIG"] = self.ipi_docker_config_dir
+        ipi_docker_config_file = os.path.join(self.ipi_docker_config_dir, "config.json")
+        shutil.copy(self.docker_config_file, ipi_docker_config_file)
```

Comment:
`self.docker_config_file` if the path to docker config, why not just set it directly to `DOCKER_CONFIG` 

```--docker-config-file PATH           Path to Docker config.json file.
                                      File must include token for `registry.ci.openshift.org`
                                      (Needed only for AWS IPI clusters)

```

Reply:
@myakove since `DOCKER_CONFIG` is a directory where `config.json` should exists, and user can pass `self.docker_config_file` either as a absolute path, or just file name (exists at $PWD), so instead of adding conditions to get dir path for these, I just moved it to temp dir.
Hope this explains this reason.

Reply:
I don't see any reason to create temp dir and copy the file just to avoid `conditions`, the user sends us the path, use it.

Reply:
updated code